libsplinter/src/network/connection_manager/messages.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crossbeam_channel::Sender;
+
+#[derive(Debug)]
+pub enum PmMessage {
+    Shutdown,
+    Connect(String, Sender<Response>),
+    Disconnect(String),
+    Subscribe(String, Sender<Notification>),
+    UnSubscribe(String),
+    Request(Request),
+}
+
+#[derive(Debug)]
+pub struct Request {
```

Comment:
Is there a more specific name for this? Request is a tad overloaded.

Reply:
Do you have a name in mind? I was thinking of prefixing everything with "Cm" for connection manager. So `Request` would become `CmRequest`. Alternatively if you import it somewhere where the name would be ambiguous you could import it as `connection_manager::Request`.

Reply:
I suppose if they are not public api messages, `Cm` is fine.  Not a huge fan of those style of abbreviations, but eh.